fluidreview/api.py

```diff
@@ -139,3 +150,45 @@ def request(self, method, url_suffix, **kwargs):
             self.session = self.initialize_session()
             response = getattr(self.session, method)(urljoin(BASE_API_URL, url_suffix), **kwargs)
         return response
+
+
+def process_user(fluid_user):
+    """
+    Create/update User and Profile model objects based on FluidReview user info
+
+    Args:
+        fluid_user (ReturnDict): Data from a fluidreview.serializers.UserSerializer object
+    """
+    user = User.objects.filter(email=fluid_user['email']).first()
+    if not user:
+        user = User.objects.create(email=fluid_user['email'], username=fluid_user['email'])
+    profile, _ = Profile.objects.get_or_create(user=user)
+    if not profile.fluidreview_id:
+        profile.fluidreview_id = fluid_user['id']
+        profile.name = fluid_user['full_name']
+        profile.save()
+
+
+def parse_webhook(webhook):
+    """
+    Attempt to load a WebhookRequest body as JSON and assign its values to other attributes.
+
+    Args:
+        webhook (WebhookRequest): WebhookRequest instance
+
+    """
+    try:
+        body_json = json.loads(smart_text(webhook.body))
+        for attr in get_field_names(WebhookRequest):
+            if attr in body_json:
+                if attr in ('award_cost', 'amount_to_pay'):
+                    if body_json[attr]:
+                        setattr(webhook, attr, Decimal(body_json[attr]))
+                else:
+                    setattr(webhook, attr, body_json[attr])
+        webhook.status = WebhookParseStatus.SUCCEEDED
+    except (json.JSONDecodeError, InvalidOperation) as exc:
```

Comment:
It seems like we want to include all exceptions here, since we would want to set `FAILED` for those cases

Reply:
OK